cirq/contrib/routing/utils.py

```diff
@@ -74,3 +77,20 @@ def is_valid_routing(
                                                    can_reorder=can_reorder)
     logical_operations = swap_network.get_logical_operations()
     return cca.is_topologically_sorted(circuit_dag, logical_operations, equals)
+
+
+def get_circuit_connectivity(circuit: 'cirq.Circuit') -> nx.Graph:
```

Comment:
It would be helpful to annotate the edges with their multiplicity and/or depth in the circuit (i.e., which time slice they would be in), but that can wait for a later PR.

Reply:
yeah out of scope. It would take some thought on how to present that information visually